src/disputables/index.js

```diff
@@ -0,0 +1,261 @@
+ * @param {String} disputableAddress Address of the disputable app where the disputed action is taking place
+ * @param {String} disputableActionId Disputed action's id relative to the disputable app (e.g. in the context of a disputable voting app, vote #6)
+ * @returns {Object} Object containing the disputed action short and long description as well as the URL.
+ */
+export async function describeDisputedAction(
+  disputeId,
+  organization,
+  disputableAddress,
+  disputableActionId
+) {
+  try {
+    if (cachedDescriptions.has(disputeId)) {
+      return cachedDescriptions.get(disputeId)
+    }
+    // Get disputable appId
+    const disputableContract = getContract(disputableAddress, disputableAbi)
+    const appId = await disputableContract.appId()
+
+    if (DISPUTABLE_ACTIONS.has(appId)) {
+      const { abi, entity, fn, scriptPosition } = DISPUTABLE_ACTIONS.get(appId)
+
+      // Get disputed action script
+      const disputableAppContract = getContract(disputableAddress, abi)
+      const result = await disputableAppContract[fn](disputableActionId)
+      const evmScript = result[scriptPosition]
+
+      // Get long and short description corrsponding to the disputed action.
+      const [
+        disputedActionRadspec,
+        disputedActionText,
+      ] = await describeActionScript(evmScript, organization)
+
+      const disputedActionDescription = {
+        disputedActionText,
+        disputedActionRadspec,
+
+        // Build URL where the disputed action is taking place
+        disputedActionURL: buildDisputedActionUrl(
+          organization,
+          disputableAddress,
+          entity,
+          disputableActionId
+        ),
+      }
+
+      // Cache disputed action description
+      cachedDescriptions.set(disputeId, disputedActionDescription)
+
+      return disputedActionDescription
+    }
+  } catch (err) {
+    console.error('Error describing disputable action', err)
+  }
+
+  return { disputedActionRadspec: 'No description' }
+}
+
+/**
+ * Describes the disputed action script using @aragon/connect
+ * @param {String} evmScript The EVM script to describe
+ * @param {String} organization Address of the organization in question
+ * @returns {Array} Array of items where the first item is an array of transaction requests in the path of the disputed action
+ *                    and the second item, the app belonging to the organization where the disputed action is taking place.
+ */
+async function describeActionScript(evmScript, organization) {
+  const org = await connect(organization, 'thegraph', {
+    chainId: env('CHAIN_ID'),
+  })
+  const apps = await org.apps()
+
+  const transactionRequests =
+    (await describeScript(evmScript, apps, org.provider)) || []
+
+  // In order to get the terminal action/s we must search through the `transactionRequests` children (if any)
+  const terminalActions = getTerminalActions(transactionRequests)
```

Comment:
I wonder if we should re-organize the rest of the function a bit:

- Bail early if `terminalActions` is empty
- Push `disputedActionText` and `terminalAction` together, since the comment about the first action only applies to `disputedActionText`

Reply:
Much better